_posts/en/newsletters/2021-04-07-newsletter.md

```diff
@@ -44,36 +44,28 @@ BOLTs][bolts repo].*
 
 - [Bitcoin Core #20286][] rpc: deprecate `addresses` and `reqSigs` from rpc outputs FIXME:Xekyo
 
-- [Bitcoin Core #20197][] improves the diversity of peer connections by updating
-  Bitcoin Core's inbound peer eviction logic to protect the longest-running
-  onion peers from eviction. It also adds deterministic unit test coverage for
-  the current eviction protection logic.
-
-  Changes in this area are delicate, as extreme care must be taken to avoid
-  opening the node to attacker-triggered network partitioning. The peer eviction
-  logic selects an inbound peer to evict after filtering out (protecting) peers
-  having desirable, difficult-to-forge characteristics. Onion peers have
-  historically been disadvantaged under these eviction criteria due to their
-  higher latency relative to IPv4 and IPv6 peers, which meant they would be
-  evicted more often, degrading the diversity of peer connections and leading to
-  users filing issues like [Bitcoin Core #11537][] and [Bitcoin Core #19500][].
-
-  An initial response to the issue was made by [Bitcoin Core #19670][], which
-  began reserving protected slots for localhost peers as a proxy for inbound
-  onion peers. However, manually configured Tor hidden services not using
-  `-bind=addr[:port]=onion` could not be detected as inbound onion connections
-  until more recently with the merge of [Bitcoin Core #19991][] that added a
-  specific port for them.
-
-  With the new logic, half of the protected spots are allocated to onion peers
-  (if any), sorted by longest uptime, even if they're not longest uptime
-  overall. Any remaining slots of the half are then allocated to protect
-  localhost peers (if any), sorted by longest uptime.
-
-  Now that support for the I2P privacy network has been added to Bitcoin Core
-  for the upcoming release (see [Newsletter #139][news139 i2p]), a next step
-  will be to extend eviction protection to I2P peers, as they generally have
-  higher latency than onion peers.
+- [Bitcoin Core #20197][] improves the diversity of peer connections by
+  updating Bitcoin Core's inbound peer eviction logic to protect the
+  longest-running onion peers from eviction. It also adds deterministic
+  unit test coverage for the current eviction protection logic.  Onion
+  peers have historically been disadvantaged due to their higher latency
+  relative to IPv4 and IPv6 peers, leading to users filing issues like
+  [Bitcoin Core #11537][] and [Bitcoin Core #19500][].
```

Comment:
I think this sentence belongs with the next paragraph, not this one. Also, the edits removed context that I think is helpful to understanding this sentence. If this description is too long, I'd suggest slimming down details further down, like in the second sentence of the second paragraph about `-bind`.

Reply:
I think this description is still a bit long and could benefit from being slimmed down a little more. @jonatack - can you cut out the bits about `-bind` and try to get this down to two paragraphs?

Reply:
Done.

Reply:
(@harding thanks for adding references in the summary to our previous coverage of PRs 19670 and 19991!)